repository_service_tuf_worker/repository.py

```diff
@@ -557,80 +653,54 @@ def bootstrap(
-        bit_length = int(
-            log(tuf_settings["services"]["number_of_delegated_bins"], 2)
-        )
-        # Succinct delegated roles (`bins`)
-        succinct_roles = SuccinctRoles([], 1, bit_length, BINS)
-        targets.signed.delegations = Delegations(
-            keys={}, succinct_roles=succinct_roles
-        )
-        # Initialize all succinct delegated roles (`bins`), update expire,
-        # sign, add to `Snapshot` meta and persist in the backend storage
-        # service.
-        for delegated_name in succinct_roles.get_roles():
-            targets.signed.add_key(
-                SSlibKey.from_securesystemslib_key(
-                    self._key_storage_backend.get(public_key).key_dict
-                ),
-                delegated_name,
-            )
-            bins_role = Metadata(Targets())
-            self._bump_expiry(bins_role, BINS)
-            self._sign(bins_role)
-            snapshot.signed.meta[f"{delegated_name}.json"] = MetaFile(
-                version=bins_role.signed.version
-            )
-            self._persist(bins_role, delegated_name)
+        if signed:
+            self._persist(root, Root.type)
+            self.write_repository_settings("ROOT_SIGNING", None)
+            self._bootstrap_online_roles(root)
+            self.write_repository_settings("BOOTSTRAP", task_id)
+            signature = "complete"
+            message = f"Bootstrap finished {task_id}"
+            logging.info(message)
 
-        # Create all Target Roles in the database RSTUFTargetRoles
-        db_target_roles = [
-            targets_schema.RSTUFTargetRoleCreate(
-                rolename=target_role, version=1
+        else:
+            signature = "pending"
+            self.write_repository_settings(
+                f"{Root.type.upper()}_SIGNING", root.to_dict()
             )
-            for target_role in succinct_roles.get_roles()
-        ]
-        targets_crud.create_roles(self._db, db_target_roles)
-
-        # Update `Snapshot` meta with targets
-        snapshot.signed.meta[f"{Targets.type}.json"] = MetaFile(
-            version=targets.signed.version
-        )
-
-        # Update expire, sign and persist the top level roles (`Targets`,
-        # `Timestamp``, `Snapshot`) in the backend storage service.
-        for role in [targets, snapshot, timestamp]:
-            self._bump_expiry(role, role.signed.type)
-            self._sign(role)
-            self._persist(role, role.signed.type)
+            self.write_repository_settings("BOOTSTRAP", f"signing-{task_id}")
+            message = f"Root version {root.signed.version} is pending sign"
+            logging.info(message)
 
         result = ResultDetails(
             status="Task finished.",
             details={
-                "bootstrap": True,
+                "bootstrap": signed,
+                "signature": signature,
+                "message": message,
```

Comment:
Everything in RSTUF is async.
When the user starts the bootstrap, it will generate a task id.
The user uses the endpoint to check the task status, and we try to bring as many details as possible.
I will simplify to bootstrap and message only. I think there is enough info.